evalml/tests/component_tests/test_oversampler.py

```diff
@@ -262,18 +264,36 @@ def test_samplers_perform_equally(
     np.testing.assert_equal(sorted(y_im), expected_y)
 
 
-def test_smoten_categorical_boolean(X_y_binary, im):
+@pytest.mark.parametrize("use_nullable", [True, False])
+def test_smoten_categorical_boolean(X_y_binary, im, use_nullable):
     X, y = X_y_binary
-    X.ww.set_types({0: "Categorical", 1: "Boolean"})
-    X = X.drop(range(2, len(X.columns)), axis=1)
+    bool_ltype = "BooleanNullable" if use_nullable else "Boolean"
+    X.ww.set_types({0: "Categorical"})
+    X.ww[len(X.columns)] = ww.init_series(
+        pd.Series([True, False] * 50),
```

Comment:
This test is trying to artificially add Categorical and Boolean columns to `X_y_binary` which is all float columns. The `Boolean` transformation works, but `BooleanNullable` does not allow converting from numeric values that aren't 1s or 0s. So I changed this test to set one of the float columns to be Categorical and add in new Boolean column of actual boolean values. Similar changes had to be made to `test_smotenc_boolean_numeric` for the same reasons.